internal/datastore/postgres/migrations/zz_migration.0009_add_xid8_columns.go

```diff
@@ -0,0 +1,46 @@
+package migrations
+
+import (
+	"context"
+
+	"github.com/jackc/pgx/v4"
+)
+
+const addTransactionXIDColumns = `
+	ALTER TABLE relation_tuple_transaction
+		ADD COLUMN xid xid8 NOT NULL DEFAULT (pg_current_xact_id()),
+		ADD COLUMN snapshot pg_snapshot NOT NULL DEFAULT (pg_current_snapshot())`
+
+const addTupleXIDColumns = `
+	ALTER TABLE relation_tuple
+		ADD COLUMN created_xid xid8 NOT NULL DEFAULT (pg_current_xact_id()),
+		ADD COLUMN deleted_xid xid8 NOT NULL DEFAULT ('9223372036854775807');`
+
+const addNamespaceXIDColumns = `
+	ALTER TABLE namespace_config
+		ADD COLUMN created_xid xid8 NOT NULL DEFAULT (pg_current_xact_id()),
+		ADD COLUMN deleted_xid xid8 NOT NULL DEFAULT ('9223372036854775807');`
+
+const backfillTransactionXids = `
```

Comment:
do we understand the impact of this operation in a very large dataset?
How can we guarantee this does not cause an availability issue in the database while executed?
Why only backfilling the transactions table and not theother tables?
When `SELECT`ing, is the default not used on rows that already existed before the migration?

Reply:
> do we understand the impact of this operation in a very large dataset?
The transaction table is being constantly garbage collected, so this table shouldn't be super huge. I haven't measured the impact on a very large database though.

> How can we guarantee this does not cause an availability issue in the database while executed?
We don't currently.

> Why only backfilling the transactions table and not theother tables?
Transaction table needs to have unique values before we can make it the primary key. The idea here is that after you enable the "writes to both" phase, you wait at least one garbage collection period before moving on so that everything that might be selected on becomes fresh enough due to the garbage collector. These backfilled rows can't be used as revisions because they don't have an accurate snapshot value.

> When SELECTing, is the default not used on rows that already existed before the migration?
We will be using values that are new enough after waiting the garbage collection period on the actual relationship rows. E.g. all of the alive ones will be considered alive as of the time of migration.

Reply:
Maybe we should do it in chunks?

Reply:
I considered it, but we don't want to overwrite any XIDs that are actually generated by the system. The other option would be to select out the XID for the transaction that is going to be applied as the default, and then update only the rows that have that as their XID, but that requires a "go migration" and since this table is likely small for all users I didn't think it was worth the effort.

Reply:
Let's verify that GC is working and then this should be fine

Reply:
GC is working.